VAMobile/src/App.tsx

```diff
@@ -188,10 +189,9 @@ export const AuthGuard: FC = () => {
   }, [isVoiceOverTalkBackRunning, dispatch, currNewState])
 
   useEffect(() => {
-    // only run on app load
     dispatch(sendUsesLargeTextAnalytics())
     dispatch(sendUsesScreenReaderAnalytics())
-  }, [dispatch])
+  }, [dispatch, screenReaderEnabled])
```

Comment:
Shouldn't `screenReaderEnabled` only be added to this array if it's used within the `useEffect()`?

Reply:
It looks like `screenReaderEnabled` is needed in the array in order for analytics to be sent each time the screen reader is turned off/on, but Dylan can probably confirm that. Although, with this update, we'd be sending large text analytics every time the screen reader is turned off/on which isn't ideal.  We may need to move `sendUsesLargeTextAnalytics` into its own effect to prevent duplicate large text analytics from being sent

Reply:
@narin @theodur That is correct it needs to be sent each time the screen reader is turned on/off so you have to add it to the array to trigger as dispatch doesn't change.

SendUsesLargeTextAnalytics has the same issue as screen reader in that it is only sent on app launch. So it needs to be included as the same issues apply and I don't see a good hook that we can use to send that separately.

Reply:
I think Theo is implying you make a second useEffect here and get a usesLargeText trigger for that second clause so we aren't sending usesLargeTextAnalytics anytime they turn off/on the screenreader? Is that accurate Theo?

Reply:
@dumathane Yup, that's spot on. Large text analytics probably shouldn't be sent whenever a user enables/disables the screen reader, because it's unrelated. Having it in its own hook would ensure it's only sent when necessary, even if that is only when the app is first launched. Maybe the [useFontScale](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/b06a68ccfe27d1f9538fc879ce1e1964aafe221e/VAMobile/src/utils/hooks/index.tsx#L66-L80) would be a good hook to use for the new useEffect's dependency array, since it'd tell us when the font scale changes. 

Although, it looks like large text analytics are out of the scope of this ticket? It's still might be worth using `useFontScale` since large text analytics will need its own effect anyway